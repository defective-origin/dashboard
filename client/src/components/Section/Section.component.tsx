import React from 'react'

// ---| core |---
import { cn } from 'tools'

// ---| pages |---
// ---| screens |---
// ---| components |---
import Layout, { LayoutProps } from 'components/Layout'
import Actions, { ActionsProps } from 'components/Actions'
import Content from 'components/Content'
import Header from 'components/Header'
import Text from 'components/Text'

// ---| self |---
import css from './Section.module.scss'

export type SectionProps = LayoutProps & {
  title?: React.ReactNode
  actions?: ActionsProps['items']
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <Section />
 */
export function Section(props: SectionProps): JSX.Element {
  const { title, actions, children, className, ...otherProps } = props
  const _className = cn(css.Section, className)

  return (
    <Layout className={_className} stretch {...otherProps}>
      <Header className={ css.Header }>
        <Text className={ css.Title } content={title} v='h3' />

        <Actions className={ css.Actions } items={actions} />
      </Header>

      <Content className={ css.Content }>
        {children}
      </Content>
    </Layout>
  )
}

Section.displayName = 'Section'

export default Section
