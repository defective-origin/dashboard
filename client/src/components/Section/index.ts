/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './Section.component'

/* INJECT_EXPORT_PLACE */
export * from './Section.component'
