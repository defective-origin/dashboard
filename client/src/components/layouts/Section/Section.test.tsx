// ---| tests |---
import { render } from 'tests'

// ---| self |---
import Section from './Section.component'

describe('[Section] component', () => {
  it('should render component', () => {
    const container = render(<Section />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
