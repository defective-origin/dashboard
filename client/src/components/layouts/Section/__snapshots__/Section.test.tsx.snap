// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[Section] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="layout item stretch _Section_44cb7f"
  >
    <div
      class="block block--x item _Header_f4f7a7 _Header_44cb7f"
      style="align-items: center; grid-area: top;"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignLeft text text--md _Title_44cb7f css-zy7tjt-MuiTypography-root"
      />
      <div
        class="block block--x item _Actions_dd10e3 _Actions_44cb7f"
        style="align-items: center;"
      />
    </div>
    <div
      class="layout item g--xxs _Content_99ea91 _Content_44cb7f"
      style="grid-area: center;"
    />
  </div>
</DocumentFragment>
`;
